### PR TITLE
OPENEUROPA-2288: Integrate country vocabulary into address module

### PR DESCRIPTION
**Pull request by brummbar**

### CODE REVIEW

File: tests/src/Kernel/CorporateCountriesRdfKernelTestBase.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_corporate_countries\Kernel;
+
+use Drupal\Tests\rdf_entity\Kernel\RdfKernelTestBase;
+use Drupal\Tests\rdf_skos\Traits\SkosImportTrait;
+
+/**
+ * Abstract class for kernel tests with a test RDF country vocabulary.
```
Comment (imanoleguskiza) [inline tests/src/Kernel/CorporateCountriesRdfKernelTestBase.php:11]: "Abstract class for kernel tests with a test using the RDF country vocabulary."

File: modules/oe_corporate_countries_address/tests/src/Kernel/CountryRepositoryTest.php [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    ConfigurableLanguage::createFromLangcode('it')->save();
+  }
+
+  /**
+   * Tests that retrieval of country lists in different locales.
+   */
+  public function testGetList(): void {
+    $repository = $this->container->get('address.country_repository');
+
+    // The test country RDF contains 4 countries, but one is not mapped in the
```
[the remark this answers is not in the document]
Comment (brummbar) [reply modules/oe_corporate_countries_address/tests/src/Kernel/CountryRepositoryTest.php:46]: Ops, old test comment, indeed needs to be fixed.

File: modules/oe_corporate_countries_address/src/Repository/CountryRepository.php [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      return $this->definitions[$locale];
+    }
+
+    $cache_key = 'oe_corporate_countries_address.op_countries.' . $locale;
+    if ($cached = $this->cache->get($cache_key)) {
+      $this->definitions[$locale] = $cached->data;
+    }
+    else {
+      $this->definitions[$locale] = $this->doLoadDefinitions($locale);
+      $this->cache->set($cache_key, $this->definitions[$locale], CacheBackendInterface::CACHE_PERMANENT, ['countries']);
```
[the remark this answers is not in the document]
Comment (brummbar) [reply modules/oe_corporate_countries_address/src/Repository/CountryRepository.php:87]: But then if the address module code invalidates the tag in some scenarios, our list won't be invalidated. That's why I kept the same.

File: src/Plugin/Validation/Constraint/DeprecatedCorporateCountryConstraint.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * )
+ */
+class DeprecatedCorporateCountryConstraint extends Constraint {
+
+  /**
+   * The constraint message.
+   *
+   * @var string
+   */
+  public $message = 'The country "%value" is deprecated. Please specify a replacement.';
```
[the remark this answers is not in the document]
Comment (brummbar) [reply src/Plugin/Validation/Constraint/DeprecatedCorporateCountryConstraint.php:24]: Yeah I tried to keep the same as the address module in here. Loading the entity name at this point is also quite "expensive" code wise.

File: modules/oe_corporate_countries_address/src/Repository/CountryRepository.php [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    $entities = $storage->loadMultiple(array_keys($countries));
+
+    $definitions = [];
+    foreach ($entities as $id => $entity) {
+      $translation = $this->entityRepository->getTranslationFromContext($entity, $locale);
+      $definitions[$countries[$id]['alpha-2']] = $translation->label();
+    }
+
+    uasort($definitions, function (string $a, string $b) use ($locale): int {
+      return $this->transliteration->transliterate($a, $locale) <=> $this->transliteration->transliterate($b, $locale);
```
[the remark this answers is not in the document]
Comment (brummbar) [reply modules/oe_corporate_countries_address/src/Repository/CountryRepository.php:121]: That is true for some things in the frontend, but when searching amongst 250+ countries in a select you want to speed up the search alphabetically. There's no "political" reason in here, just practical one. I used the transliteration to order properly special characters like `Å` in an even more natural way.

File: tests/src/Kernel/CorporateCountriesRdfKernelTestBase.php (hunk 1 above)
Comment (brummbar) [reply tests/src/Kernel/CorporateCountriesRdfKernelTestBase.php:11]: "test RDF country vocabulary" or "RDF country test vocabulary". I mean that this test has a test example vocabulary provided.

File: modules/oe_corporate_countries_address/src/Repository/CountryRepository.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (imanoleguskiza) [reply modules/oe_corporate_countries_address/src/Repository/CountryRepository.php:87]: I checked and it doesn't seem to be the case, but other sites may be doing it so agreed, lets keep it the same.

File: tests/src/Kernel/CorporateCountriesRdfKernelTestBase.php (hunk 1 above)
Comment (imanoleguskiza) [reply tests/src/Kernel/CorporateCountriesRdfKernelTestBase.php:11]: Gotcha

File: modules/oe_corporate_countries_address/src/Repository/CountryRepository.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (imanoleguskiza) [reply modules/oe_corporate_countries_address/src/Repository/CountryRepository.php:121]: Agreed.

**Pull request merged**